manifests/mirror.pp

```diff
@@ -85,22 +78,28 @@
 
     $exec_aptly_mirror_create_require = [
       Package['aptly'],
-      File['/etc/aptly.conf'],
       Exec["aptly_mirror_gpg-${title}"],
     ]
   } else {
     $exec_aptly_mirror_create_require = [
       Package['aptly'],
-      File['/etc/aptly.conf'],
     ]
   }
 
-  $cmd_options_string = join(reject(join_keys_to_values(merge($default_cmd_options, $cmd_options), '='), '.*=$'), ' ')
-  $cmd_string         = rstrip("${aptly_cmd} create ${cmd_options_string} ${title} ${location} ${release} ${components}")
+  $cli_options_string = join(reject(join_keys_to_values($cli_options, '='), '.*=$'), ' ')
+  $cmd_string         = rstrip("${aptly_cmd} create ${cli_options_string} ${title} ${location} ${release} ${components}")
+
+  # Since the create and show commands don't share a common set of
+  # options, we need to extract the config
+  if has_key($cli_options, '-config') {
+    $config_string = "-config=${cli_options['-config']}"
```

Comment:
Took me a minute to grok what this was about. Maybe naming it `$unless_config_string` would make it immediately clear that it's pulling out the config option for use in the `unless` case. Same would apply down below in `repo.pp` & `snapshot.pp`